### ISSUE TITLE
Tor Browser 10.0.2 fails lie detection (length test)

### ISSUE THREAD

**Issue by abrahamjuliot** [abrahamjuliot/creepjs#83]
```js
HTMLCanvasElement.getContext:
// Expected length 1 and got 2 
```

**Comment by Thorin-Oakenpants** [comment]
Can you elaborate (ELI5)?

**Comment by abrahamjuliot** [comment]
The [length](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length) property of a function object returns the number of arguments the function expects. 

`getContext` has 2 parameters, `contextType` and `contextAttributes`, but only expects `contextType`. 

`HTMLCanvasElement.prototype.getContext.length` should return 1. Linux and Mac return 1 as expected, but Windows returns 2. 🤔

**Comment by Thorin-Oakenpants** [comment]
Right: so is that caused by a pref, or by a TB patch? Or by being in PB Mode, or by RFP?

**Comment by abrahamjuliot** [comment]
Correction: not Windows only

This appears to be trigger by setting the security level to safer. RFP and private browsing mode do not seem to affect this.

**Comment by Thorin-Oakenpants** [comment]
This sounds like readPixels and webGL? On `standard` webgl is blocked, period. On safest the API is enabled and it becomes click to play. I can't think of anything else in the slider that would cause that - AFAIK, it's controlled (partly at least) via NoScript

The error messages generated by readPixel also provide entropy, which is probably the equivalency of what you're seeing here

**Comment by abrahamjuliot** [comment]
The invalid `HTMLCanvasElement.prototype.getContext.length` appears to be caused by NoScript (which also fails a number of prototype lie tests).
